setup/aio/010_Install_Puppet_Agent.rb

```diff
@@ -1,10 +1,23 @@
 test_name "Install Packages" do
 
   dev_builds_url  = ENV['DEV_BUILDS_URL'] || 'http://builds.delivery.puppetlabs.net'
+  nightly_builds_url = ENV['NIGHTLY_BUILDS_URL'] || 'http://nightlies.puppet.com'
+
+  sha = ENV['SHA']
+  release_stream = ENV['RELEASE_STREAM'] || 'puppet'
 
   step "Install puppet-agent..." do
-    sha = ENV['SHA']
-    install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{sha}/artifacts/#{sha}.yaml", hosts)
+    # If SHA='latest', then we're installing from nightlies
+    if sha == 'latest'
+      opts = {
+        :release_yum_repo_url => nightly_builds_url + '/yum',
+        :release_apt_repo_url => nightly_builds_url + '/apt',
+        :puppet_collection => "#{release_stream}-nightly"
+      }
+      install_puppet_agent_on(hosts, opts)
+    else
+      install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{sha}/artifacts/#{sha}.yaml", hosts)
```

Comment:
I added a helper to wrap this recently (so that the artifactory doesn't need to be specified) if that helps: https://github.com/puppetlabs/beaker-puppet/commit/fef05c36fe53fecc01482ba08c93ee8217682bed

Reply:
This is why I should have been paying more attention to the changes y'al made to this code base! That's p sweet

Reply:
@caseywilliams I'm curious about this helper you wrote

beaker already has a utility to check the accessibility of a url that gets called in `install_from_build_data_url`. Is there a reason you added a brand new check with `dev_builds_accessible_on?`? The main thing I can see is if you're running tests on a host that doesn't have access to builds.d.pl.net versus the local system that's running beaker.

The other thing is that you can't pass in a custom url. I know it's super unlikely, but there is a possibility that folks have duplicated how we've set up builds.d.pl.net with the yaml file. Since we're already giving people a means to customize that url, I'm hesitant to take that functionality away

Reply:
I intended for the dev builds accessibility helper to be a good-enough check that would determine whether the hosts can access resources within the corporate VPN, regardless of whether you were actually trying to download something at the time (like for printing human-friendly error messages), but now that you put it that way, I realize the name is bad and it's probably not even that useful :/

This PR does look good to me

Reply:
It is helpful, and it does exactly what you would expect based on the name!

Mostly I noticed beaker has the method `link_exists?` at https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/helpers/web_helpers.rb#L20. I was curious why you weren't using that

Reply:
Ah yeah, I got some feedback on the original PR that encouraged checking all of the SUTs' access as opposed to just the controlling host